### DISCUSSION TITLE
Clear bookmarks in a file automatically when it is closed

### DISCUSSION THREAD

**Discussion by tolache** [alefragnani/vscode-bookmarks#550] [Q&A] [answered]
Hi! Thanks for creating the extension. Can I configure the extension to clear the bookmarks in a file automatically when it is closed?

**Comment by alefragnani** [comment, marked as answer]
Hi @tolache ,

There is no setting in the extension for such behavior. The bookmarks are always persisted, and recovered, when you reopen any file.

Could you provide more details about your use case. Why would you want the bookmarks to be removed when you close the file?

**Reply by tolache** [reply]
Hi @alefragnani, thanks for the reply.
Sure, here's how I use it.

1. Open file A and add some bookmarks (<kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>K</kbd>).
2. Cycle the bookmarks a few times (<kbd>Ctrl</kbd>+<kbd>Alt</kbd>+<kbd>L</kbd>).
3. Close file A and open file B.
4. Add some bookmarks.
5. Cycle through the bookmarks.

Expected behavior:
On step 5, it only cycles through the bookmarks added to file B.

Actual behavior:
On step 5, when I reach the last bookmark in file B and keep cycling, it reopens file A and shows me a bookmark in file A.

**Reply by alefragnani** [reply]
So, you don’t really need _clear_ the bookmarks, instead _navigate only in the current file_.

Maybe you should try using the `bookmarks.navigateThroughAllFiles` setting, with `false` value. It will keep cycling only in the current file only

Hope this helps

**Reply by tolache** [reply]
You are right, this is what I need. Thank you for your help 👍
